### ISSUE TITLE
How to handle state management for each model item in a collection?

### ISSUE THREAD

**Issue by SKeeneCode** [xamantra/momentum#13]
I'm unfamiliar with flutter/dart/widgets so bear with me if this is a silly question.

I have a `Book` model, which contains a collection of `Page` models.
The `Book` controller has all the logic for adding/deleting pages and so on.

I want the `Page` controller to have the logic for adding/removing text in the page. I'm not sure how to achieve this with Momentum. I can write the Page controller, but it has a 1 to 1 relationship with a single `Page` object right?

Basically I want the user to be able to select a page, and then edit that page. In the past when I used a MVVM architecture in TornadoFX I could simply change the ViewModels item field to whatever Page Model the user clicked on and then when the ViewModel changed, it updated the model behind the scenes.

Can I achieve something similar with Momentum, or will I be fighting against its convention?

**Comment by xamantra** [comment]
[Listify](https://github.com/xamantra/listify) is a perfect example of this.
It has a nested list model structure. The same with what you mention about `Book` having a list of `Page` models. I highly recommend you take a look at its source code.

> `but it has a 1 to 1 relationship with a single Page object right?`

Yes, but because momentum allows you to manipulate other controllers using the `dependOn<T>()` method, you can easily update the `Page` and `Book` model.

### Here's a real snippet from [Listify](https://github.com/xamantra/listify):
#### InputController
```dart
  void submit() {
    if (!validInput()) {
      triggerAction(
        actionMessage: "List name and items are both required.",
        action: InputAction.ErrorOccured,
      );
      return;
    }
    if (exists()) {
      triggerAction(
        actionMessage: "List data with same list items or same list name already exist.",
        action: InputAction.ErrorOccured,
      );
      return;
    }
    var _listController = dependOn<ListController>();
    if (model.editingList) {
      _listController.updateList(model.editListName, model.listName, model.items);
      triggerAction(action: InputAction.ListDataEdited);
    } else {
      _listController.addList(ListData(listName: model.listName, items: model.items));
      triggerAction(action: InputAction.ListDataAdded);
    }
    reset(clearHistory: true);
  }
```

#### ListController
```dart
  void updateList(String listName, String newListName, List<ListItem> newItems) {
    var items = List<ListData>.from(model.items);
    var index = items.indexWhere((x) => x.listName == listName);
    if (index != -1) {
      items.removeAt(index);
      items.insert(index, ListData(listName: newListName, items: newItems));
      model.update(items: items);
      dependOn<CurrentListController>().viewData(model.items[index]);
    }
  }
```

- `InputController` and `ListController` are separate momentum controller.
- The input controller calls a method from list controller: `_listController.updateList(...)`.
- Then the list controller's `updateList()` method finds the list that is being edited and update it with the new values provided in the parameters.

**Comment by SKeeneCode** [comment]
Meant to close this a while ago but forgot, thanks for the excellent explanation! Helped me figure it out :)
